### DISCUSSION TITLE
Asking for otel-auto-instrumentation-java agent

### DISCUSSION THREAD

**Discussion by SusanaG97** [metalbear-co/mirrord#2201] [Q&A] [answered]
Hi. I'm trying to run my java application on IntelliJ, with the use of the mirrord plugin, and I'm getting this error:

`Picked up JAVA_TOOL_OPTIONS:  -javaagent:/otel-auto-instrumentation-java/javaagent.jar
Error opening zip file or JAR manifest missing : /otel-auto-instrumentation-java/javaagent.jar
Error occurred during initialization of VM
agent library failed to init: instrument`

I have downloaded this java agent and set the JAVA_TOOL_OPTIONS to its location (correctly, I can verify by echo in any terminal). But seems it's not sticking here, as the location shown on the error doesn't match.

This is my mirrord config:

{
    "feature": {
        "env": {
            "override": {
            }
        },
        "network": {
            "dns": true,
            "incoming": {
                "mode": "steal"
            },
            "outgoing": true
        },
        "fs": "read"
    },
    "target": {
        "path": {
            "deployment": "broker"
        },
        "namespace": "runtime"
    }
}

Any help would be really appreciated!

**Comment by aviramha** [comment]
Hey,
It's probably picking it up from the remote, use this config
```json
{
"feature": {
"env": {
"override": {
}
},
"network": {
"dns": true,
"incoming": {
"mode": "steal"
},
"outgoing": true
},
"fs": "read", "local": ["/otel.+"]
},
"target": {
"path": {
"deployment": "broker"
},
"namespace": "runtime"
}
}
```
or try this
```json
{
"feature": {
"env": {
"override": {
}
},
"network": {
"dns": true,
"incoming": {
"mode": "steal"
},
"outgoing": true
},
"fs": "read", "local": ["/$"]
},
"target": {
"path": {
"deployment": "broker"
},
"namespace": "runtime"
}
}
```

We read jars locally by default, but it could be that it tries to list the directory first, which is resolved remotely (`'/`)
I'd put the otel binary in a different directory that can be marked safely as local (since marking `/` local might have other implications)

**Reply by SusanaG97** [reply]
Thank you!
I've tried both configs and the error remains. I've also placed the required agent jar in /home and it made no difference

**Reply by aviramha** [reply]
it's quite weird - would you like to schedule to investigate it together?
https://calendly.com/aviram-5/30min

**Reply by aviramha** [reply, marked as answer]
ahh I think I know what happens - it has this in the remote pod and tries to load it locally.
You can just exclude that env from being fetched with this config:
```json
{
"feature": {
"env": {
"override": {
},
"exclude": ["JAVA_TOOL_OPTIONS"]
},
"network": {
"dns": true,
"incoming": {
"mode": "steal"
},
"outgoing": true
},
"fs": "read"
},
"target": {
"path": {
"deployment": "broker"
},
"namespace": "runtime"
}
}
```

**Reply by SusanaG97** [reply]
setting JAVA_TOOL_OPTIONS here:
"env": {
            "override": {
                "JAVA_TOOL_OPTIONS": ""
            }
        }
allows me to at least run the application
